src/bgw/job.c

```diff
@@ -425,13 +426,8 @@ ts_bgw_job_get_share_lock(int32 bgw_job_id, MemoryContext mctx)
 											&got_lock);
 	if (job != NULL)
 	{
-		if (!got_lock)
-		{
-			/* since we blocked for a lock , this is an unexpected condition */
-			ereport(ERROR,
-					(errcode(ERRCODE_INTERNAL_ERROR),
-					 errmsg("could not acquire share lock for job=%d", bgw_job_id)));
-		}
+		/* since we blocked for a lock , this is an unexpected condition */
+		ASSERT_OR_ERROR(got_lock, "could not acquire share lock for job=%d", bgw_job_id);
```

Comment:
while this is useful, this is due to some other session holding the lock. More useful to give more detailed info about the lock holder than an Assert condition